File: dcp-solutions/2024-12-18/solution.php
```php
function longestConsecutive($nums) {
    $num_set = array_flip($nums);
    $max_length = 0;

    foreach ($num_set as $num => $value) {
        if (!isset($num_set[$num - 1])) {
            $current_num = $num;
            $current_length = 1;

            while (isset($num_set[$current_num + 1])) {
                $current_num += 1;
                $current_length += 1;
            }

            $max_length = max($max_length, $current_length);
        }
    }

    return $max_length;
}
```

File: dcp-solutions/2024-12-18/solution.c
```c
int longest_consecutive(int* nums, int numsSize) {
    if (numsSize == 0) return 0;
    unordered_set<int> num_set(nums, nums + numsSize);
    int max_length = 0;

    for (int num : num_set) {
        if (num_set.find(num - 1) == num_set.end()) {
            int current_num = num;
            int current_length = 1;

            while (num_set.find(current_num + 1) != num_set.end())
                current_num += 1, current_length++;

            max_length = max(max_length, current_length);
        }
    }

    return max_length;
}
```